_adr/90/index.adoc

```diff
@@ -0,0 +1,135 @@
+For example, if the kafka id is `12345` then Kafka boostrap host will have the form of `foobar-xsqhyyza.12345.kafka.bf2.dev`.
+b. When a `developer` kafka is assigned to a given data plane, all routes created will be under the `*.trial.kafka.bf2.dev` domain. 
+For example, if the kafka id is `12345` then Kafka boostrap host will have the form of `foobar-sqfsazk.trial.kafka.bf2.dev`.
+
+3. No migration to be performed for Kafkas.
+The proposed solution will be applied to only new Kafkas.  
+Existing Kafkas will continue to share the wildcard certificate on the `*.kafka.bf2.dev` domain.  
+The shared certificate for the already existing Kafkas will now be automatically renewed by the fleet manager.
+
+4. Certificate renewals will also be handled automatically by the fleet manager using the https://github.com/caddyserver/certmagic[certmagic library]. 
+Renewals will be blast radius aware to renew certificates in small runs on per Kafka basis.
+The intention is to reduce the risk of compromising all our certificates in case there is an outage or an issue with the renewal system in Letsencrypt. 
+It also removes the chances of bombarding Letsencrypt with bulk requests or rollout of all the Kafkas in one big batch after renewals. 
+
+5. Revocation will be handled per Kafka instance.
+It will follow a GitOps approach where SRE will request to revoke a certificate for a Kafka.
+Once the compromised certificate has been revoked, a new certificate for that Kafka will be generated.
+For already existing Kafkas, they'll continue to share the newly generated certificate: see proposal number (3). 
+
+6. When a standard Kafka is removed from the fleet manager, its corresponding certificate will also be revoked. 
+This is with an excpetion for all Kafkas that uses a shared certificate i.e Kafkas that existed before the solution is in place or all new developer Kafkas that uses a shared wildcard `*.trial.kafka.dev` certificate.
+
+7. There will be separate Letsencrypt's account per kas-fleet-manager environment. 
+For each of these acccounts, a request to overrides the various https://Letsencrypt.org/docs/rate-limits[Letsencrypt limits] will be done by following https://Letsencrypt.org/docs/rate-limits/#a-id-overrides-a-overrides[Letsencrypt limits overrides].
+There are two main limits that we should be aware of and are of interests: (1) Certificates per Registered Domain which has a default of 50 certificates per week (2) You can create a maximum of 300 New Orders per account per 3 hours.
+Each of these two limits needs to be bumped to a bigger value after capacity planning for certificate generation has been done.
+
+NOTE: There is an existing logic in the data plane that ensures that all sensitive parts in the ManagedKafka CR are not written in plain text. 
+Instead a secret is created and only references to the secrets will be added before the ManagedKafka is created in the data plane.
+
```

Comment:
Does the dataplane logic already handle renewal and revocation? If so, let's say so explicitly.

In particular, what has the logic for knowing about the impending passing of a certificate's `notAfter` (bearing in mind that the certificate in question might be a root or intermediate CA cert, though I'd expect Let's Encrypt would be avoiding issuing end entity certificates with `notAfter` after any CA certificate in the chain).

Reply:
For Managed Kafka, the data plane components are only responsible for stripping out the certificate from the MK CR and creating a secret out of it.
Certificate management is handled entirely in the fleet manager and it is responsible for automatic generation, renewal (within 1 month before the certificate expires and revocation - when a managed instance is removed or via an admin endpoint, once we detect that there are certificates that have been compromised)